src/modules/10-common/components/ModeSelector/__tests__/ModeSelector.test.tsx

```diff
@@ -0,0 +1,250 @@
+  SSCA: {
+    icon: 'ssca-main',
+    label: 'common.sscaText',
+    homePageUrl: '/account/px7xd_BFRCi-pfWPYXVjvw/module/ssca?noscope=true',
+    shouldVisible: true,
+    hasLicense: false,
+    color: '--default-module-border',
+    backgroundColor: '',
+    shortLabel: 'common.sscaShortLabel',
+    moduleIntro: 'common.moduleIntro.softwareSupplyChainAssurance'
+  },
+  IDP: {
+    icon: 'idp',
+    label: 'common.purpose.idp.fullName',
+    homePageUrl: '/account/px7xd_BFRCi-pfWPYXVjvw/module/idp/default',
+    shouldVisible: true,
+    hasLicense: false,
+    color: '--idp-border',
+    backgroundColor: '',
+    shortLabel: 'common.purpose.idp.name',
+    moduleIntro: 'common.moduleIntro.idp'
+  },
+  CET: {
+    icon: 'cet',
+    label: 'common.purpose.cet.continuous',
+    homePageUrl: '/account/px7xd_BFRCi-pfWPYXVjvw/module/cet?noscope=true',
+    shouldVisible: true,
+    hasLicense: false,
+    color: '--cet-border',
+    backgroundColor: '',
+    shortLabel: 'common.purpose.errorTracking.title',
+    moduleIntro: 'common.moduleIntro.continuousErrorTracking'
+  },
+  SEI: {
+    icon: 'sei-main',
+    label: 'common.purpose.sei.fullName',
+    homePageUrl: '/account/px7xd_BFRCi-pfWPYXVjvw/module/sei?noscope=true',
+    shouldVisible: false,
+    hasLicense: false,
+    color: '--default-module-border',
+    backgroundColor: '',
+    shortLabel: 'common.purpose.sei.continuous',
+    moduleIntro: 'common.moduleIntro.insights'
+  }
+}
+
+jest.mock('framework/PreferenceStore/PreferenceStoreContext')
+;(usePreferenceStore as jest.Mock).mockImplementation(() => {
+  return {
+    setPreference: jest.fn(),
+    preference: {
+      orderedModules: ['CD', 'CI', 'CV', 'CF', 'CE', 'STO', 'CHAOS', 'CODE', 'SSCA', 'IDP', 'CET'],
+      selectedModules: ['CD', 'CI', 'CV', 'CF', 'CE', 'STO', 'CHAOS', 'CODE', 'SSCA', 'IDP', 'CET']
+    },
+    clearPreference: jest.fn()
+  }
+})
+jest.mock('@common/hooks/useNavModuleInfo', () => {
+  return {
+    __esModule: true,
+    default: jest.fn().mockImplementation(() => {
+      return {}
+    }),
+    useNavModuleInfoMap: () => {
+      return navModuleInfoMapMock
+    },
+    DEFAULT_MODULES_ORDER: ['CD', 'CI', 'CV', 'CF', 'CE', 'STO', 'CHAOS', 'CODE', 'SSCA', 'IDP', 'CET']
+  }
+})
+
+describe('Mode selector', () => {
+  const checkModuleCard = (container: HTMLElement, moduleLabel: string, href: string): void => {
+    const moduleCard = queryByText(container, moduleLabel) as HTMLElement
+    expect(moduleCard).toBeInTheDocument()
+    expect(moduleCard.closest('a')).toHaveAttribute('href', href)
+  }
+
+  test('should render mode selector', async () => {
+    const { container } = render(
+      <TestWrapper defaultFeatureFlagValues={{ CDS_NAV_2_0: true }}>
+        <ModeSelector />
+      </TestWrapper>
+    )
+    const modeSelectorButton = container.querySelector('[data-icon="nine-dot-options"]') as HTMLInputElement
+    await act(async () => {
+      fireEvent.click(modeSelectorButton)
+    })
```

Comment:
there's a cleaner+safer way to do clicks:

```suggestion
    const user = userEvent.setup()
    await user.click(modeSelectorButton)
```

Reply:
Noted. Will do it this way in my subsequent PRs